src/App.vue

```diff
@@ -17,51 +17,41 @@
 </template>
 
 <script setup lang="ts">
-import qs from 'qs';
-
-import HeaderVue from './components/Header.vue';
-import FooterVue from './components/Footer.vue';
-
-import {projectConfigurationPath} from './Utilities/SelectedProject';
-
-import {
-  languageCodes,
-  languageCodesLowerCase,
-} from './Utilities/LanguageCodes';
-import { onBeforeMount, onMounted, ref, watch } from 'vue';
+import { computed } from '@vue/reactivity';
+import { onBeforeMount, provide, ref } from 'vue'
 import { useI18n } from 'vue-i18n';
 import { useRoute, useRouter } from 'vue-router';
-import { getProjectIdFromEnvironment, getProjectIdFromCookies } from './Client';
 
-const infoMessageText = ref('');
+import { Client, getEnvironmentIdFromCookies, getEnvironmentIdFromEnvironment } from './Client';
+import FooterVue from './components/Footer.vue';
+import HeaderVue from './components/Header.vue';
+import { type LanguageCode,languageCodes, languageCodesLowerCase } from './Utilities/LanguageCodes'
+import {projectConfigurationPath} from './Utilities/SelectedProject';
+import { ClientKey } from './Utilities/Symbols';
 
 const i18n = useI18n({ useScope: 'global' });
 const language = i18n.locale.value;
 const router = useRouter();
 const route = useRoute();
+const client = ref(Client);
 
+const infoMessageText = computed<string>(() => route.query['infoMessage'] as string);
```

Comment:
The proper type would be `string | undefined`, because you can't be sure that the query is present, can you?